agent/api/task/task.go

```diff
@@ -760,16 +764,51 @@ func (task *Task) isNetworkModeVPC() bool {
 	return true
 }
 
-func (task *Task) addNetworkResourceProvisioningDependency(cfg *config.Config) {
+func (task *Task) addNetworkResourceProvisioningDependency(cfg *config.Config) error {
 	if !task.isNetworkModeVPC() {
-		return
+		return nil
 	}
 	pauseContainer := apicontainer.NewContainerWithSteadyState(apicontainerstatus.ContainerResourcesProvisioned)
 	pauseContainer.TransitionDependenciesMap = make(map[apicontainerstatus.ContainerStatus]apicontainer.TransitionDependencySet)
 	pauseContainer.Name = NetworkPauseContainerName
 	pauseContainer.Image = fmt.Sprintf("%s:%s", cfg.PauseContainerImageName, cfg.PauseContainerTag)
 	pauseContainer.Essential = true
 	pauseContainer.Type = apicontainer.ContainerCNIPause
+
+	// Set pauseContainer user the same as proxy container user
+	if task.GetAppMesh() != nil {
+		appMeshConfig := task.GetAppMesh()
+		for _, container := range task.Containers {
+			if container.Name == appMeshConfig.ContainerName {
```

Comment:
you can get rid of a layer of indent here if you do:

```
if containerName != appmesh... {
    continue
}

Reply:
Done.